patcher.sh

```diff
@@ -0,0 +1,302 @@
+		mv "$SCRIPT_DIR/AM2R.AppDir/bin/icon.png" "$OUTPUT/icon.png"
+		if [ "$SYSTEMWIDE" = false ]; then
+			mv "$SCRIPT_DIR/AM2R.AppDir/bin/AM2R.desktop" "$desktopPath"
+		fi
+		rm -R "$SCRIPT_DIR/AM2R.AppDir"
+
+	elif [ "$OSCHOICE" = "android" ]; then
+		echo "Creating an APK for Android..."
+		apktoolPath="$SCRIPT_DIR/utilities/android/apktool.jar"
+		uberPath="$SCRIPT_DIR/utilities/android/uber-apk-signer.jar"
+
+		echo "Applying Android patch..."
+		xdelta3 -dfs "$OUTPUT/data.win" "$SCRIPT_DIR/data/droid.xdelta" "$OUTPUT/game.droid"
+
+		cp "$SCRIPT_DIR/data/android/AM2RWrapper.apk" "$SCRIPT_DIR/utilities/android/"
+		rm "$OUTPUT/D3DX9_43.dll" "$OUTPUT/AM2R.exe" "$OUTPUT/data.win"
+		mv "$OUTPUT" "$SCRIPT_DIR/utilities/android/assets"
+
+		if [ -f "$SCRIPT_DIR/data/android/AM2R.ini" ]; then
+			cp -p "$SCRIPT_DIR/data/android/AM2R.ini" "$SCRIPT_DIR/utilities/android/assets/"
+		fi
+
+		echo "Installing new datafiles..."
+		cp -R "$SCRIPT_DIR"/data/files_to_copy/* "$SCRIPT_DIR/utilities/android/assets"
+
+		if [ "$HQMUSIC" = true ]; then
+			cp "$SCRIPT_DIR"/data/HDR_HQ_in-game_music/*.ogg "$SCRIPT_DIR/utilities/android/assets/"
+		fi
+
+		echo "Packaging APK..."
+		echo "If Java JDK 8 or newer is not present, this will fail!"
+		# decompile the apk
+		java -jar "$apktoolPath" d -f "$SCRIPT_DIR/utilities/android/AM2RWrapper.apk"
+		cp -R "$SCRIPT_DIR/utilities/android/assets/" "$SCRIPT_DIR/AM2RWrapper/"
+		echo "Editing apktool.yml..."
+	    sed -i "s/doNotCompress:/doNotCompress:\n- ogg/" "$SCRIPT_DIR/AM2RWrapper/apktool.yml"
+		# build apk
+		java -jar "$apktoolPath" b "$SCRIPT_DIR/AM2RWrapper" -o "$SCRIPT_DIR/AM2RWrapper.apk"
+		echo "Signing APK..."
+		java -jar "$uberPath" -a "$SCRIPT_DIR/AM2RWrapper.apk"
+		# Cleanup
+		rm -R "$SCRIPT_DIR/AM2RWrapper.apk" "$SCRIPT_DIR/utilities/android/AM2RWrapper.apk" "$SCRIPT_DIR/utilities/android/assets/" "$SCRIPT_DIR/AM2RWrapper/"
+		# Move APK
+		mv "$SCRIPT_DIR/AM2RWrapper-aligned-debugSigned.apk" "$SCRIPT_DIR/AndroidM2R_"$VERSION"-signed.apk"
+
+	else
+		echo "Invalid OS \"$OSCHOICE\"! Cannot patch anything!"
+		return 1
+	fi
+
+	echo ""
+	echo "The operation was completed successfully. See you next mission!"
+	return 0
+}
+
+# ----
+# Main function starts here
+# ---
+
+echo "-------------------------------------------"
+echo ""
+echo "AM2R ${VERSION} Shell Autopatching Utility"
+echo "Scripted by Lojemiru and Miepee"
+echo ""
+echo "-------------------------------------------"
+
+if (( $# <= 0 )); then
+	APPIMAGE=true
+	AM2RZIP="$SCRIPT_DIR/AM2R_11.zip"
+	input=""
+	echo "Select your patch type:"
+	echo ""
+	echo "1 - Linux"
+	echo "2 - Android"
+	echo ""
+	echo "Awaiting input:"
+	read -n1 input
+	echo ""
+	if [ $input = "1" ]; then
+		OSCHOICE="linux"
+	elif [ $input = "2" ]; then
+		OSCHOICE="android"
+	else
+		echo "Invalid OS!"
+		exit 1
+	fi
+
+	echo "Install high quality in-game music? Increases filesize by 194 MB!"
+	echo "[y/n]"
+	read -n1 input
+	echo ""
+	if [ $input = "y" ]; then
```

Comment:
Same as above. Doing `${input,,}` will also make this case insensitive, which is a bit nicer.
```suggestion
	if [[ "${input,,}" = "y" ]]; then
```

Reply:
Applied locally.